sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +246,97 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       .addOrderBy("assessment.submittedDate", "DESC")
       .getMany();
   }
+
+  /**
+   * Creates a manual assessment for the application.
+   * @param applicationId application id.
+   * @param userId user id who triggered the manual reassessment.
+   */
+  manualReassessment(applicationId: number, note: string, userId: number) {
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const currentUser = { id: userId } as User;
+      const now = new Date();
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      const application = await applicationRepo.findOne({
+        select: {
+          id: true,
+          currentAssessment: true as unknown,
+          student: { id: true },
+        },
+        relations: [
+          "currentAssessment",
+          "currentAssessment.offering",
+          "currentAssessment.studentAppeal",
+          "student",
+        ],
+        where: {
+          id: applicationId,
+        },
+      });
+
+      if (!application) {
+        throw new CustomNamedError(
+          `Application not found.`,
+          APPLICATION_NOT_FOUND,
+        );
+      }
+      if (
+        application.currentAssessment.studentAssessmentStatus !==
+        StudentAssessmentStatus.Completed
+      ) {
+        throw new CustomNamedError(
+          `Application current assessment expected to be '${StudentAssessmentStatus.Completed}' to allow manual reassessment.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      if (application.isArchived) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment after being archived.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+
+      // Adds application note.
```

Comment:
Please use the `createStudentNote` already in place.
https://github.com/bcgov/SIMS/blob/1e08be02c368cb17ec55b8ace0218a920d477990/sources/packages/backend/libs/services/src/note/note.shared.service.ts#L21